lib/plugins/wallet/cardano/mnemonic.js

```diff
@@ -0,0 +1,64 @@
+const fs = require('fs')
+const cardanoApi = require('./client')
+const {sha256} = require('js-sha256')
+
+const coinUtils = require('../../../coin-utils')
+const options = require('../../../options')
+
+const cryptoRec = coinUtils.getCryptoCurrency('ADA')
+const configPath = coinUtils.configPath(cryptoRec)
+
+function getCardanoMnemonic() {
+  return fs
+    .readFileSync(options.cardanoMnemonicPath, 'utf8')
+    .trim()
+    .split(/[ \r\n]+/)
+}
+
+function getCardanoMnemonicHashed() {
+  return sha256(getCardanoMnemonic().join(' '))
+}
+
+function getWalletConfiguration() {
+  return JSON.parse(fs.readFileSync(configPath))
+}
+
+function saveWalletConfiguration(configuration) {
+  fs.writeFileSync(configPath, JSON.stringify(configuration))
+}
+
+async function walletId() {
+  let configuration = getWalletConfiguration()
```

Comment:
There can be an exception throw e.g. by reading the file, or invalid content in JSON.parse. Do you want to catch it and also throw some meaningful error?

Reply:
yes, because most of them are fatal for us... but I will be happy to discuss about that